### ISSUE TITLE
[Bug]: Metrics error for --num-scheduler-steps > 1

### ISSUE THREAD

**Issue by yuleil** [vllm-project/vllm#8233]
### Your current environment

Your output of `python collect_env.py` here

### 🐛 Describe the bug

When `--num-scheduler-steps=n`, `do_log_stats()` is called only once every `n` step. However, when calculating `num_generation_tokens` within `do_log_stats()`, it only adds one generation token for each seq_group. This results in the `Avg generation throughput` metrics output being incorrect.

![image](https://github.com/user-attachments/assets/30072995-8ea2-4029-a65d-7936ff9fa9c5)

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by youkaichao** [comment]
contribution welcome!

**Comment by wuisawesome** [comment]
I can probably take a look at this later this week/next week if no one gets to it before then.
